frontend/src/Hooks/useWorkshops.ts

```diff
@@ -22,13 +20,11 @@ export default function useWorkshops(props: UseWorkshopsProps) {
     }
     function search(){
         if(searchTerm !== "") {
-            setIsSearch(true)
-            setSearchResults(props.workshops
-                .filter(workshop =>
-                    (workshop.services.filter(service =>
-                        service.toLowerCase().includes(searchTerm.toLowerCase())).length>0) ||
-                    (workshop.inventory.filter(component =>
-                        component.category.toLowerCase().includes(searchTerm.toLowerCase())).length>0))) 
+            axios.get("api/workshops/search/" + searchTerm)
+                .then(r => r.data)
+                .then(results => setSearchResults(results))
+                .then(() => setIsSearch(true))
+                .catch((error) => console.error(error))
```

Comment:
Nice move to the backend! Is it considered better practice to search in the backend? (when there are no data privacy concerns that prohibit frontend search in the first place)

I thought it could be better to have it in the frontend to save precious backend computing power, but since you deliberately moved it there you probably did some research? Would love to hear your opinion on it!

Reply:
In my specific case, searching in the frontend would work. But since I know that my app will become a big success with more than 1000 registered workshops, fetching all of those to the frontend would cost a lot of performance. This is why I decided to let the backend do the search and just fetch the searchResults as needed.